libraries/state_history/include/eosio/state_history/log.hpp

```diff
@@ -419,6 +420,23 @@ class state_history_log {
       ilog("${name} regeneration complete", ("name", index.display_path()));
    }
 
+   void check_log_and_index_on_init() {
+      if(log.size() == 0)
+         return;
+
+      try {
+         log_header first_header = log.unpack_from<decltype(first_header)>(0);
```

Comment:
is the `decltype` needed? Why not just `log.unpack_from<log_header>`?

Reply:
yeah that's what something like clang-tidy will suggest. But generally I don't like using `auto` where unnecessary, especially for such a simple case

Reply:
Personally I would prefer just `log_header first_header = log.unpack_from<log_header>(0);`
I don't see the point of changing it as it matches other usages in the file.